src/frontend/App.tsx

```diff
@@ -22,12 +22,14 @@ import {QueryClient, QueryClientProvider} from '@tanstack/react-query';
 import {ObservationProvider} from './contexts/ObservationsContext';
 import {MessagePortLike} from './lib/MessagePortLike';
 import {createMapeoClient} from '@mapeo/ipc';
-import {ApiLoading} from './sharedComponents/ApiLoading';
+import {ServerLoading} from './ServerLoading';
 import {ActiveProjectProvider} from './contexts/ProjectContext';
+import {initializeNodejs} from './initializeNodejs';
 
 const queryClient = new QueryClient();
 const messagePort = new MessagePortLike();
 const mapeoApi = createMapeoClient(messagePort);
+initializeNodejs();
```

Comment:
what happens in the app if this throws? we may need to catch the error and depending on how the app responds, do something to improve the ux in the case of failure

Reply:
Yeah, I think we should plan out where we want to put error boundaries (although this technically would not be caught by an error boundaries since it is outside the react component, so I don't actually know the best way to handle this).

Reply:
To do this:

```suggestion
function noop () {}
const nodejsInitPromise = initializeNodejs();
// make sure we catch so we don't have an uncaught error - but we want to store the original promise (above) so we can check if it rejected later
nodejsInitPromise.catch(noop)
```

Then in a component:
```js
React.useEffect(() => {
  nodejsInitPromise.then(() => {
    setState('initialized')
  }).catch(e => {
    setState('error')
  })
}, []) // <-- no need for dep, since function def is not inside component
```

Can show an error based on state, or render rest of the app once initialized.